pinot-query-planner/src/main/java/org/apache/pinot/query/planner/QueryPlan.java

```diff
@@ -39,13 +44,15 @@
 public class QueryPlan {
   private final List<Pair<Integer, String>> _queryResultFields;
   private final Map<Integer, StageNode> _queryStageMap;
-  private final Map<Integer, StageMetadata> _stageMetadataMap;
+  private final Map<Integer, WorkerMetadata> _stageMetadataMap;
+  private final Map<Integer, DispatchablePlanMetadata> _dispatchablePlanMetadataMap;
```

Comment:
- 1st map is for distributedStagePlan. one can argue it doesn't need to be materialized until distributeStagePlan is constructed
- 2nd map is for dispatcher only - to dispatch stage to physical server.

nit: rename _stageMetdataMap to _workerMetadataMap